### ISSUE TITLE
Improved encoding/decoding?

### ISSUE THREAD

**Issue by mtwilliams** [fredwu/simple_bayes#4]
Heya!

Loving this library, except I've encountered an unfortunate pathological case.

I have trained a classifier on roughly 288,000 labeled texts. Cardinality of labels is 5, and length of the text ~5 words.

Here's how it's configured:

``` elixir
model:          :bernoulli,
storage:        :file_system,
file_path:      @storage,
default_weight: 1.0,
smoothing:      0.0,
stem:           false,
stop_words:     []
```

I then persisted this trained model to disk. It's roughly 11Mb when stored. Then I attempted to load it back into memory...

After 10 (!) minutes I received this error:

```
** (ArithmeticError) bad argument in arithmetic expression
    (stdlib) :math.pow(7318, 24158)
             lib/simple_bayes/classifier/models/bernoulli.ex:7: SimpleBayes.Classifier.Model.Bernoulli.probability_of/3
             lib/simple_bayes/classifier/probability.ex:37: anonymous fn/4 in SimpleBayes.Classifier.Probability.for_collection/3
    (elixir) lib/map.ex:114: Map.do_new_transform/3
             lib/simple_bayes/classifier.ex:13: SimpleBayes.Classifier.classify/3
```

Ouch.

So, there's two things here:

  1) There's an issue prevent me from reloading the trained classifier.
  2) Loading is painfully slow.

When I have the time, I'll explore the reasons behind (1). For now, I'll see how far `dets` can get me.

**Comment by fredwu** [comment]
@mtwilliams Thanks for reporting this issue, it's always nice to have feedback on real world usage. Let me know what you can find, I'll do some testing on my end too. :)

**Comment by fredwu** [comment]
I did some preliminary testing and it appears this is due to Erlang's `:math` library using floats for the `pow/2` operation. Using an integer version implemented in Elixir resolves this issue, but still when dealing with large number of items the float calculations will overflow, I'll see what can be done.

**Comment by fredwu** [comment]
I've just pushed a fix. Could you please update your mix dependency to use the github version and see if that resolves your issue?

**Comment by fredwu** [comment]
Just pushed another commit that added `dets` as a storage option, please give it a try and see how it effects the performance. :)

**Comment by fredwu** [comment]
Closing this for now as I've released v0.10.0 with Dets support, please feel free to re-open this issue if the problem persists. :)
